### PR TITLE
feat: add support for mock vault

### PR DESCRIPTION
**Pull request by Niraj-Kamdar**
**Changes:**
- Using Mock vault in dev environment
- No need to fork while developing (forking is only used for testing)
- Auto minting to `ADMIN_ADDRESS` while starting node while on dev env.

### CODE REVIEW

File: hardhat/hardhat.config.ts [hunk 1]
```diff
@@ -151,4 +151,9 @@ const config: HardhatUserConfig = {
   },
 };
 
+if (process.env.TEST) {
```
Comment (levity) [inline hardhat/hardhat.config.ts:154]:
```suggestion
if (process.env.TEST || process.env.CI) {
```

File: .env.example [hunk 2]
```diff
@@ -12,6 +12,7 @@ REACT_APP_API_BASE_URL=http://localhost/api
 
 # for Hardhat
 ETHEREUM_RPC_URL=https://mainnet.infura.io/v3/YOUR_KEY
+ADMIN_ADDRESS=0x0
```
Comment (exrhizo) [inline .env.example:15]: There is a vault owner and a circle admin in the contracts - rename this to be more specific
`HH_OWNER_ADDRESS` for example

File: hardhat/scripts/deploy/00-mock-yearn/03-mock-yvyfi-deploy.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  });
+
+  // Note: We aren't using the MockVaultFactory.createVault() method here because
+  // deploymentInfo won't track any transactions that create new contracts.
+  // So, we are manually deploying the Vault and registering it with the MockRegistry.
+  const yvYFI = await deploy('yvYFI', {
+    contract: 'MockVault',
+    from: deployer,
+    args: [yfi.address, 'YFI yVault', 'yvYFI'],
+  });
```
Comment (exrhizo) [inline hardhat/scripts/deploy/00-mock-yearn/03-mock-yvyfi-deploy.ts:35]: These deploy scripts aren't in alignment with the engineering principle of `Don't Repeat Yourself`

File: hardhat/package.json [hunk 4]
```diff
@@ -7,15 +7,17 @@
...
     "deploy": "hardhat deploy --network localhost",
-    "test": "hardhat test"
+    "test": "TEST=1 hardhat test"
   },
   "files": [
     "dist"
   ],
   "devDependencies": {
     "@nomiclabs/hardhat-ethers": "^2.0.2",
+    "@nomiclabs/hardhat-ganache": "^2.0.1",
```
Comment (exrhizo) [inline hardhat/package.json:20]: We pin dependencies,
https://betterdev.blog/pin-exact-dependency-versions/

**Review by exrhizo — CHANGES_REQUESTED** [review]

**Review by exrhizo — COMMENTED** [review]
@levity would be in a better place to approve this - although the tests are failing in github, so that is something to fix

**Review by levity — APPROVED** [review]

**Pull request merged**